### PR TITLE
rename specs tests

### LINKED ISSUE #63: Rename specs as tests
# 📋 Info

`spec(s)` is an old Jasmine name for tests. This is confusing. We should call the files tests instead.

# 🏁 DoD

- [ ] rename files `*.spec.js*`->`*.test.js*`
- [ ] rename directory `specs`->`tests` (or `test`)
- [ ] update path in `package.json` etc. so that linting and tests work.

### PR DESCRIPTION
**Pull request by skymandr**
This renames the "specs" to "tests".

Also fixes a few warnings and  errors that occurred when running the tests, but which don't fail the pipeline, so please check the files.

Resolves #63

### CODE REVIEW

**Comment by skymandr** [comment]
So, the remaining problem is with `error` in `<PolynomialConstructionContainer/>`, which should be a `string` according to the proptypes, but is set from


```javascript
  handleConstruction() {
    const { id, accessToken } = this.props.cccMetadata;
    const { degreeOfPolynomial } = this.state;
    return API.SetNewCalibrationPolynomial(id, degreeOfPolynomial, accessToken)
      .then(this.handleConstructionResults)
      .catch(this.handleConstructionResultsError);
  }

  handleConstructionResultsError(reason) {
    this.setState({ error: reason });
  }
```

Letting it be a string in the tests provokes the linter, but this could be disabled. Not sure it is a good thing that it assumes that it is a string though, so I suggest:

```javascript
  handleConstructionResultsError(reason) {
    this.setState({ error: reason.message ?? reason });
  }
```
or similar.

What do you say, @local-minimum ?

**Comment by local-minimum** [comment]
> So, the remaining problem is with `error` in `<PolynomialConstructionContainer/>`, which should be a `string` according to the proptypes, but is set from
> 
> ```js
>   handleConstruction() {
>     const { id, accessToken } = this.props.cccMetadata;
>     const { degreeOfPolynomial } = this.state;
>     return API.SetNewCalibrationPolynomial(id, degreeOfPolynomial, accessToken)
>       .then(this.handleConstructionResults)
>       .catch(this.handleConstructionResultsError);
>   }
> 
>   handleConstructionResultsError(reason) {
>     this.setState({ error: reason });
>   }
> ```
> 
> Letting it be a string in the tests provokes the linter, but this could be disabled. Not sure it is a good thing that it assumes that it is a string though, so I suggest:
> 
> ```js
>   handleConstructionResultsError(reason) {
>     this.setState({ error: reason.message ?? reason });
>   }
> ```
> 
> or similar.
> 
> What do you say, @local-minimum ?

Hard to say without the lint error when fixed. Reasonably reason can be a `string`, `error` or `undefined` in the most general case... so it would be `string | undefined` after your suggestion.

If it's used as a string in the actual component then yes fix it, but attempt to fix the lint issue too rather than ignore

File: scanomatic/ui_server_data/js/ccc/components/PolynomialResultsColonyHistogram.jsx [hunk 1]
```diff
@@ -106,10 +106,18 @@ export default class PolynomialResultsColonyHistogram extends React.Component {
...
         />
-        <span className="poly-colony-txt">{valueFormatter(independentMeasurement, 2)} cells</span>
+        <span className="poly-colony-txt">{valueFormatter(independentMeasurements, 2)} cells</span>
       </div>
     );
   }
 }
 
-PolynomialResultsColonyHistogram.propTypes = CCCPropTypes.colonyShape;
+PolynomialResultsColonyHistogram.propTypes = {
```
Comment (local-minimum) [inline scanomatic/ui_server_data/js/ccc/components/PolynomialResultsColonyHistogram.jsx:115]: was it only used here or why this change?

File: scanomatic/ui_server_data/js/ccc/containers/PolynomialConstructionContainer.jsx [hunk 2]
```diff
@@ -20,10 +20,8 @@ export default class PolynomialConstructionContainer extends React.Component {
     };
 
     this.handleConstruction = this.handleConstruction.bind(this);
-    this.handleConstructionResults = this.handleConstructionResults
-      .bind(this);
-    this.handleConstructionResultsError =
-            this.handleConstructionResultsError.bind(this);
+    this.handleConstructionResults = this.handleConstructionResults.bind(this);
```
Comment (local-minimum) [inline scanomatic/ui_server_data/js/ccc/containers/PolynomialConstructionContainer.jsx:23]: the functions ought not to be bound here but instead be arrow functions on the class, but that's probably better a separate issue.

**Review by local-minimum — CHANGES_REQUESTED** [review]

File: scanomatic/ui_server_data/js/ccc/containers/PolynomialConstructionContainer.jsx (hunk 2 above)
Comment (skymandr) [reply scanomatic/ui_server_data/js/ccc/containers/PolynomialConstructionContainer.jsx:23]: Sure, I don't understand the problem though, so I'd appreciate if you create it or at least commit to fleshing it out if I do.

File: scanomatic/ui_server_data/js/ccc/components/PolynomialResultsColonyHistogram.jsx (hunk 1 above)
Comment (skymandr) [reply scanomatic/ui_server_data/js/ccc/components/PolynomialResultsColonyHistogram.jsx:115]: This is a reversal of a change from a previous PR, which didn't quite work (gave the weird error ```ERROR: 'Warning: Failed prop type: The prop `isRequired` is marked as required in `PolynomialResultsColonyHistogram`, but its value is `undefined`.
    in PolynomialResultsColonyHistogram'``` even though the linter didn't complain). The `colonyShape` was only used here anyway.

Comment (local-minimum) [reply scanomatic/ui_server_data/js/ccc/components/PolynomialResultsColonyHistogram.jsx:115]: then it must be removed yes.

File: scanomatic/ui_server_data/js/ccc/containers/PolynomialConstructionContainer.jsx (hunk 2 above)
Comment (local-minimum) [reply scanomatic/ui_server_data/js/ccc/containers/PolynomialConstructionContainer.jsx:23]: it's annoying that the way `this` works in those is redefined here in the constructor instead of them being defined as intended from the beginning. It has to do with javascript prototypes, so if interested google about the differences between function and arrow statement and how this affects `this`.

File: scanomatic/ui_server_data/js/ccc/components/PolynomialResultsColonyHistogram.jsx (hunk 1 above)
Comment (skymandr) [reply scanomatic/ui_server_data/js/ccc/components/PolynomialResultsColonyHistogram.jsx:114]: done

**Comment by skymandr** [comment]
The remaining annoyance is
```console
SUMMARY:
✔ 394 tests completed
ℹ 1 test skipped
```
when running tests. For the life of me I cannot work out what test is skipped...

File: scanomatic/ui_server_data/js/ccc/containers/PolynomialConstructionContainer.jsx (hunk 2 above)
Comment (skymandr) [reply scanomatic/ui_server_data/js/ccc/containers/PolynomialConstructionContainer.jsx:23]: I am not really that interested in anything except to know whether you will create the issue or if I should. :)

Comment (local-minimum) [reply scanomatic/ui_server_data/js/ccc/containers/PolynomialConstructionContainer.jsx:23]: i can make the issue

**Comment by local-minimum** [comment]
> The remaining annoyance is
> 
> ```
> SUMMARY:
> ✔ 394 tests completed
> ℹ 1 test skipped
> ```
> 
> when running `npm test`. For the life of me I cannot work out what test is skipped...

it's probably something in the test runners config that sets this... no clue what though...

**Review by local-minimum — APPROVED** [review]
